contracts/core/SmartSessionBase.sol

```diff
@@ -337,51 +339,90 @@ abstract contract SmartSessionBase is ISmartSession, NonceManager {
         return address($sessionValidators[permissionId][account].sessionValidator) != address(0);
     }
 
+    function isISessionValidatorSet(PermissionId permissionId, address account) external view returns (bool) {
+        return _isISessionValidatorSet(permissionId, account);
+    }
+
     function isPermissionEnabled(PermissionId permissionId, address account) external view returns (bool) {
         return $enabledSessions.contains(account, PermissionId.unwrap(permissionId));
     }
 
-    function isPermissionEnabled(
-        PermissionId permissionId,
-        address account,
-        PolicyData[] memory userOpPolicies,
-        PolicyData[] memory erc1271Policies,
-        ActionData[] memory actions
-    )
-        external
-        view
-        returns (bool isEnabled)
-    {
-        //if ISessionValidator is not set for permissionId, the permission has not been enabled yet
-        if (!_isISessionValidatorSet(permissionId, account)) {
-            return false;
-        }
-        bool uo = $userOpPolicies.areEnabled({
+    // This function accepts not the array of policies, but full PolicyData,
+    // So it is easier to use it with an EnableSessions object
+    // If you just need to check array of addresses, use is____PolicyEnabled methods in a loop
+    function areUserOpPoliciesEnabled(address account, PermissionId permissionId, PolicyData[] memory userOpPolicies) external view returns (bool) {
```

Comment:
is it possible to use calldata here, in case this is queried onchain?

Reply:
done